gateway_to_backend/protocol_buffers_files/config_message.proto

```diff
@@ -91,6 +99,10 @@ message SinkNewConfig {
 
     // State of sink
     optional OnOffState sink_state = 10;
+
+    // Only processed if CDD_API feature flag is set
+    // 0-length payloads are used to clear an entry
+    repeated ConfigurationDataItem configuration_data_content = 11;
```

Comment:
Is this intended to be used to set single item?

I was thinking that these operations used for managing the CDD content could warrant own messages / topics. The intention was to have one for adding, one for getting, and then one for deleting individual items (or a list of items). There should not be a need to manage the whole list of CDD items at once (instead, combining these should be done only at the sink).

Reply:
Yeap, at the low level those would be similar (all coupled to NPD/CDD when sent in Mesh). However, what I meant, at the gateway-to-backend API this now introduces a whole new operation pattern for the SinkNewConfig. None of the other messages has such, instead in all of the other cases write is "atomic" (or whatever you want to call it), i.e. newly provided value replaces the old value (instead of appending or removing parts of it).